packages/contracts-bedrock/src/cannon/interfaces/IPreimageOracle.sol

```diff
@@ -47,4 +47,19 @@ interface IPreimageOracle {
     /// @param _partOffset The offset of the preimage to read.
     /// @param _preimage The preimage data.
     function loadKeccak256PreimagePart(uint256 _partOffset, bytes calldata _preimage) external;
+
+    /// @notice Resets the caller's large pre-image metadata in preparation for beginning the absorption of a new
+    ///         large keccak256 pre-image.
+    /// @param _offset The offset of the preimage part to load into the oracle during absorbtion of the preimage.
+    /// @param _claimedSize The claimed size of the preimage.
+    function initLargeKeccak256Preimage(uint128 _offset, uint64 _claimedSize) external;
+
+    /// @notice Absorbs a part of the caller's large keccak256 pre-image.
+    /// @param _data The pre-image segment to absorb.
+    /// @param _finalize Whether or not to finalize the absorbtion process. If true, the contract will pad the data
+    ///                  passed per the pad10*1 rule.
+    function absorbLargePreimagePart(bytes calldata _data, bool _finalize) external;
```

Comment:
As I understand it:
1. The data needs to be added by calling this method in order
2. The only thing constraining the order is the sender account's nonce (and their policy on when to create/sign the next tx in the sequence)

Given how complex nonce management can get, and how much money could be wasted if data winds up landing out of order I wonder if the contracts should have a way to verify the data is in order. Could just add a param to this call that specifies the starting offset. If the starting offset doesn't match the length of data currently added bail out.